### PR TITLE
Split the classes from EventHubsTestUtils.scala in seperate files

### PR DESCRIPTION
**Pull request by Fokko**
The EventHubsTestUtils.scala was very long, therefore it is better to split the classes/objects into seperate files. I didn't change any of the content of the files.

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/eventhubs/utils/SimulatedCachedReceiver.scala [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package org.apache.spark.eventhubs.utils
```
Comment (sabeegrewal) [inline core/src/main/scala/org/apache/spark/eventhubs/utils/SimulatedCachedReceiver.scala]: Could you add the apache header to all these new files?

**Comment by sabeegrewal** [comment]
I agree this should be done - just a minor change + resolving the minor merge conflicts and I'll get this merged! Thanks for the contributions :)

File: core/src/main/scala/org/apache/spark/eventhubs/utils/SimulatedCachedReceiver.scala (hunk 1 above)
Comment (Fokko) [reply core/src/main/scala/org/apache/spark/eventhubs/utils/SimulatedCachedReceiver.scala]: Good point, I've added the headers.

**Comment by Fokko** [comment]
@sabeegrewal I've rebased onto master.

**Comment by sabeegrewal** [comment]
@Fokko thanks man :) great stuff, I appreciate it!

**Pull request merged**